src/builder/command.rs

```diff
@@ -4109,15 +4109,23 @@ impl<'help> Command<'help> {
 
         if !self.is_set(AppSettings::DisableHelpSubcommand) {
             debug!("Command::_check_help_and_version: Building help subcommand");
+            let help_about = "Print this message or the help of the given subcommand(s)";
             let mut help_subcmd = Command::new("help")
-                .about("Print this message or the help of the given subcommand(s)")
-                .arg(
-                    Arg::new("subcommand")
-                        .action(ArgAction::Append)
-                        .num_args(..)
-                        .value_name("SUBCOMMAND")
-                        .help("The subcommand whose help message to display"),
-                );
+                .about(help_about)
+                // help can take all other subcommands, including help, as a subcommand
+                .subcommands(
+                    self.get_subcommands()
+                        .filter(|sc| !sc.is_disable_help_flag_set())
+                        .map(|sc| {
+                            Command::new(sc.get_name())
```

Comment:
What is the performance impact of this?  We should now be recursively creating large swaths of subcommands in the parse case rather than more exclusively for the completions case?